client/src/services/LoginService.js

```diff
@@ -1,50 +1,49 @@
+import ErrorMessage from 'models/ErrorMessages';
+
 export default class LoginService {
   static login = (data) => {
+
     const formData = new FormData();
 
     formData.append('username', data.username);
     formData.append('password', data.password);
 
-
-    // Configure POST request options
-    const fetchOptions = {
+    const postRequestOptions = {
       method: 'POST',
       body: formData,
     };
 
+
     return new Promise(async (resolve, reject) => {
       try {
-        // Sends the POST Request
-        // Catch network errors and thow them to the catch block
+
+        /* Sends the POST Request */
         const response = await fetch(
-          'http://localhost:80/api/login',
-          fetchOptions
-        ).catch(() => {
-          throw new Error('Encountered a network error. Try again soon.');
+          `${process.env.SERVER_URI}/api/login`,
+          postRequestOptions
+        ).catch(() => { // Fails to connect to API
+          throw new Error(ErrorMessage.NetworkError);
         });
 
-        // Convert Response body to JSON
-        const result = await response.json();
+        const responseJson = await response.json();
 
-        // If the result is empty, throw a new error to the catch block
-        // TODO: make this work for  empty objs: {}, as this is 
-        //       what we are checking for anyway
-        if (!result) {
-          throw new Error('Received empty response from the server.');
+        /* Check for errors */
+        // Checking for empty response {}
+        if (!responseJson || Object.keys(responseJson).length === 0) {
+          throw new Error(ErrorMessage.EmptyResponse);
         }
 
-        // If the server returns a response with an error property,
-        // throw that error message to the catch block
-        if (result.hasOwnProperty('error')) {
-          throw new Error(result.error);
+        if (responseJson.hasOwnProperty('error')) {
+          throw new Error(responseJson.error);
         }
 
-        // Resolve the Promise
```

Comment:
Comments are good, but not if it feels wrong.
This comment is more words that the line of code itself, which is already self descriptive.